src/spacesuit.rs

```diff
@@ -18,79 +16,125 @@ pub fn prove(
     pc_gens: &PedersenGens,
     inputs: &Vec<Value>,
     outputs: &Vec<Value>,
-) -> Result<(SpacesuitProof, Vec<CompressedRistretto>), SpacesuitError> {
-    let m = inputs.len();
-    let n = outputs.len();
-
+) -> Result<
+    (
+        SpacesuitProof,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+        Vec<CommittedValue>,
+    ),
+    SpacesuitError,
+> {
     // Compute all intermediate values and add them to inputs and outputs,
     // returning raw variable assignments to be passed to the prover.
-    let all_values = compute_committed_values(inputs, outputs)?;
-
-    // Flatten the values into a list of raw scalars to compute the Pedersen commitments.
-    // TBD: would be nice if ProverCS had an API to do this per-value, so we can avoid
-    // flattening the structure of our input data and then unflattening it via `organize_values`.
-    // Bulletproofs repo issue: https://github.com/dalek-cryptography/bulletproofs/issues/218
-    let v: Vec<Scalar> = all_values.iter().fold(
-        Vec::with_capacity(3 * all_values.len()),
-        |mut vec, value| {
-            vec.push(value.q.into());
-            vec.push(value.a);
-            vec.push(value.t);
-            vec
-        },
-    );
+    let (tx_in, merge_in, merge_mid, merge_out, split_in, split_mid, split_out, tx_out) =
+        compute_committed_values(inputs, outputs)?;
 
-    // Make v_blinding vector using RNG from transcript
     let mut prover_transcript = Transcript::new(b"TransactionTest");
     let mut rng = {
```

Comment:
The values have low entropy, so it does not help much by mixing them into RNG here. How about (a) the blinding factors are provided externally together with open values, or (b) (for now) the `prove` method takes an RNG that we use to generate blinding factors. This way the internals do not have to be too preoccupied with blinding factors for the external input.

Reply:
done - `prove` now takes in an RNG for blinding.